packages/components/src/utils/is-deep-equal.ts

```diff
@@ -0,0 +1,32 @@
+import { isObject } from './jss';
+
+export const isDeepEqual = (newVal: unknown, oldVal: unknown): boolean => {
+  if (typeof newVal !== 'object' || typeof oldVal !== 'object') {
+    // string, boolean, number
+    return newVal === oldVal;
+  } else if (Array.isArray(newVal) || Array.isArray(oldVal)) {
+    return (
+      Array.isArray(newVal) &&
```

Comment:
That's redundant with what has been checked in the `else if` condition already?

Reply:
Yes it is, but unfortunately it seems to be needed within the return again 
![image](https://github.com/porsche-design-system/porsche-design-system/assets/109523972/c8726204-4681-4cb2-87e5-9d23c909c1ba)